src/main/java/org/andersen/homework/MainClass.java

```diff
@@ -1,69 +1,78 @@
+import java.net.URISyntaxException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.andersen.homework.exception.bus_ticket.BusTicketPriceIsNullException;
+import org.andersen.homework.exception.bus_ticket.BusTicketPriceNotEvenException;
+import org.andersen.homework.exception.bus_ticket.BusTicketStartDateIsInFutureException;
+import org.andersen.homework.exception.bus_ticket.BusTicketStartDateIsNullWhenTicketTypeIsDayWeekOrYearException;
+import org.andersen.homework.exception.bus_ticket.UndefinedBusTicketTypeException;
+import org.andersen.homework.model.entity.ticket.BusTicket;
+import org.andersen.homework.util.BusTicketFileReader;
 import org.andersen.homework.util.ValidationManager;
 
 public class MainClass {
 
-  private final static TickerService TICKER_SERVICE = new TickerService();
   private final static ValidationManager VALIDATION_MANAGER = new ValidationManager();
 
-  public static void main(String[] args) {
-    printTickets();
-    System.out.println("\n******************************************\n");
-    printUsers();
+  private static Path getPathToResource(String resourceFileName) throws URISyntaxException {
+    ClassLoader classLoader = MainClass.class.getClassLoader();
+    return Paths.get(classLoader.getResource(resourceFileName).toURI());
   }
 
-  private static void printTickets() {
-    System.out.print("Found by id: ");
-    IntStream.range(0, 999)
-        .mapToObj(i -> TICKER_SERVICE.findById((short) i))
-        .filter(Optional::isPresent)
-        .map(Optional::get)
-        .findFirst()
-        .ifPresent(System.out::println);
-
-    System.out.println("Found by stadium sector: " +
-        TICKER_SERVICE.findByStadiumSector((RandomizerUtil.getRandomFromEnum(StadiumSector.class)))
-            .stream()
-            .toList());
+  private static List<BusTicket> getBusTicketsListFromFile(String file) {
+    Path filePath = null;
+    try {
+      filePath = getPathToResource(file);
+    } catch (URISyntaxException e) {
+      System.out.println(e.getMessage());
+    }
+    assert filePath != null;
+    return BusTicketFileReader.readTicketsFromFile(
+        filePath.toAbsolutePath().toString());
+  }
 
-    Ticket limitedTicket = new Ticket((short) 123, 3.95f, "0123456789", (short) 123,
-        Boolean.TRUE, 5.5526f);
-    VALIDATION_MANAGER.validate(limitedTicket);
-    System.out.println("Limited Ticket: " + limitedTicket);
+  public static void main(String[] args) {
+    List<BusTicket> busTickets = getBusTicketsListFromFile("bus_tickets.json");
 
-    Ticket fullTicket = new Ticket((short) 123, 3.95f, "0123456789", (short) 123,
-        Boolean.TRUE, 5.5526f, LocalDateTime.now(),
-        RandomizerUtil.getRandomFromEnum(StadiumSector.class));
-    VALIDATION_MANAGER.validate(fullTicket);
-    System.out.println("Full Ticket: " + fullTicket);
+    AtomicInteger startDateViolationsCount = new AtomicInteger();
```

Comment:
I used AtomicInteger here to avoid collisions in multithreading environment in lambda function

Reply:
Do we have here multithreading environment?

Reply:
I was wrong about multithreading, but I meant that I used AtomicInteger because lambdas require effectively final variables. AtomicInteger allows state modification without needing to reassign the reference